erpnext/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -1050,7 +1050,7 @@ def make_loan_repayment_entry(self):
 			repayment_entry.save()
 			repayment_entry.submit()
 
-			loan.loan_repayment_entry = repayment_entry.name
+			frappe.db.set_value("Salary Slip Loan", loan.name, "loan_repayment_entry", repayment_entry.name)
```

Comment:
Why is `db.set_value` being preferred here? Doesn't the existing code work just fine? Why have multiple DB calls?

Reply:
No it wasn't, this function is called on submit, the doc is already validated and saved hence the field is not updated using the current code.

Reply:
Hmm, maybe you can add a `before_submit` method instead.